pkg/utils/flow/metrics.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package flow
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	registerOnce = make(chan struct{})
+
+	flowTaskTimingsDelay    *prometheus.HistogramVec
+	flowTaskTimingsDuration *prometheus.HistogramVec
+	flowTaskTotalDuration   *prometheus.HistogramVec
+	flowTaskResults         *prometheus.CounterVec
+)
+
+// Namespace is the metric namespace for the flow library.
+const Namespace = "flow"
+
+// RegisterMetrics registers the metrics for the flow library on the passed registry.
+func RegisterMetrics(r prometheus.Registerer) {
+	close(registerOnce) // Metrics can only be registered once on a registry.
+
+	factory := promauto.With(r)
+
+	// flowTaskTimingsDelay defines the histogram to record the delay in seconds from flow start until task start.
+	flowTaskTimingsDelay = factory.NewHistogramVec(
```

Comment:
Can you name the variables to match the metric names? I.e. `taskDelaySeconds` or `flowTaskDelaySeconds` for this one.